feat: add docs on custom postgres config (#14963)
supabase/supabase: apps/docs/pages/guides/platform/custom-postgres-config.mdx

import Layout from '~/layouts/DefaultGuideLayout'

export const meta = {
  id: 'custom-postgres-config',
  title: 'Custom Postgres Config',
  description: 'Configuring Postgres for your Supabase project.',
}

> Supabase Projects inclue a Postgres cluster that gets configured for optimal performance for a diverse range of workloads, using the compute add-ons being used. Custom configuration overrides can be used in order to better optimize the Postgres cluster for specific workloads that you're using.

Reviewer: ```suggestion
Supabase projects come with a Postgres cluster that is pre-configured for optimal performance. The configuration is based on a diverse range of workloads as well as the compute add-ons being used in the project. You can override this configuration to better optimize the Postgres cluster for specific workloads.
```

To get started:

1. [Install](/docs/guides/resources/supabase-cli) the Supabase CLI 1.69.0+.
1. [Log in](/docs/guides/resources/supabase-getting-started/local-development#log-in-to-the-supabase-cli) to your Supabase account using the CLI.

The `postgres config` command of the CLI can be used for setting configuration parameters:

```bash
$ supabase --experimental --project-ref <project-ref> postgres config update --config max_parallel_workers:6 --config shared_buffers:250MB
- Custom Postgres Config -
Config                  |Value    |
shared_buffers          |250MB    |
max_parallel_workers    |6        |
- End of Custom Postgres Config -
```

By default, the CLI will merge any provided config overrides with any existing ones. The `--replace-existing-overrides` flag can be used to instead force all existing overrides to be replaced with the ones being provided:

```bash
$ supabase --experimental --project-ref <project-ref> postgres config update --config max_parallel_workers:3 --replace-existing-overrides

- Custom Postgres Config -
Config                  |Value    |
max_parallel_workers    |3        |
- End of Custom Postgres Config -
```

## Considerations

> 1. The Postgres cluster will be restarted in order to change the configuration being used. This will cause momentary disruption of connections to the database; in most cases this should not take more than a few seconds.

Reviewer: ```suggestion
1. The Postgres cluster will be restarted in order to change the configuration being used. This will cause momentary disruption to existing database connections; in most cases this should not take more than a few seconds.
```

> 1. Custom Postgres Config will always override the default optimizations generated by Supabase. When change compute add-ons, this might necessitate manual updates of any relevant overrides that have been applied.

Reviewer: ```suggestion
1. Custom Postgres Config will always override the default optimizations generated by Supabase. When changing compute add-ons, this may require manual updates to any relevant overrides that have been applied.
```
